src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/StartGame.java

```diff
@@ -101,9 +267,290 @@ public StartGame(int connectionHashCode, UserMessage userMessage, Messenger mess
+        }
+        
+    }
+
+
+
+
+
     @Override
     public void react() {
-        // TODO Auto-generated method stub
+        DataDTO dataDTO = (DataDTO)userMessage.getData();
+
+        try{
+            validateData(dataDTO, database);
+
+            User user = database.getUser(dataDTO.userDTO.uuid);
+            Room room = database.getRoom(dataDTO.roomDTO.uuid);
+
+            room.startGame();
+            Game game = room.getGame();
+
+            Log.DEBUG("Game started by "+user.getName()+". Room UUID: "+room.getUuid());         
+
+            UserDataResponse userDataResponse = new UserDataResponse(user.getUuid());
+            RoomDataResponse roomDataResponse=new RoomDataResponse(room.getUuid());
+
+            TokensDataResponse tokensDataResponse=
+                new TokensDataResponse(
+                    game.getTokens(TokenType.RUBY),
+                    game.getTokens(TokenType.EMERALD),
+                    game.getTokens(TokenType.SAPPHIRE),
+                    game.getTokens(TokenType.DIAMOND),
+                    game.getTokens(TokenType.ONYX),
+                    game.getTokens(TokenType.GOLD_JOKER)
+                );
+            
+            ArrayList<Noble> nobles = game.getNobles();
+            ArrayList<NobleDataResponse> nobleDataResponses = new ArrayList<>();
+            for(Noble noble : nobles){
+                nobleDataResponses.add(new NobleDataResponse(
+                    noble.getUuid(), 
+                    noble.getPoints(), 
+                    noble.getCost(TokenType.RUBY),
+                    noble.getCost(TokenType.EMERALD),
+                    noble.getCost(TokenType.SAPPHIRE), 
+                    noble.getCost(TokenType.DIAMOND), 
+                    noble.getCost(TokenType.ONYX)
+                ));
+            }
+    
+            ArrayList<MinesCardDataResponse> firstLevelMinesCardsResponses=createMinesCardDataResponses(game.getRevealedCards(CardTier.LEVEL_1));
+            ArrayList<MinesCardDataResponse> secondLevelMinesCardsResponses=createMinesCardDataResponses(game.getRevealedCards(CardTier.LEVEL_2));
+            ArrayList<MinesCardDataResponse> thirdLevelMinesCardsResponses=createMinesCardDataResponses(game.getRevealedCards(CardTier.LEVEL_3));
+
+            UserDataResponse userToPlayResponse=new UserDataResponse(game.getCurrentPlayer().getUuid());
+                  
+            ResponseData responseData = new ResponseData(
+                userDataResponse, roomDataResponse, 
+                tokensDataResponse, nobleDataResponses, 
+                firstLevelMinesCardsResponses, secondLevelMinesCardsResponses, 
+                thirdLevelMinesCardsResponses,userToPlayResponse);
+            ServerMessage serverMessage = new ServerMessage(
+                userMessage.getContextId(), ServerMessageType.START_GAME_RESPONSE,
+                 Result.OK, responseData);
+            messenger.addMessageToSend(this.connectionHashCode, serverMessage);
```

Comment:
Your implementation sends this message only to the host of the game, but it should send it to the every user in this room.